### DISCUSSION TITLE
Customer profile available in /locate but /route always results in 'No path could be found for input', 'status_code': 400

### DISCUSSION THREAD

**Discussion by harvetech** [valhalla/valhalla#4150] [Q&A]
I am trying to add a new profile. I have gone through this commit https://github.com/valhalla/valhalla/pull/1756/files but I am still not able to have the same behavior as a taxi. The locate response looks like this 

```
'access': {'truck': False,
      **'newprofile': True,**
      'pedestrian': True,
      'wheelchair': True,
      'taxi': True,
      'HOV': False,
      'emergency': False,
      'motorcycle': False,
      'car': True,
      'moped': False,
      'bus': False,
      'bicycle': False},
```

but loki response takes forever and then it breaks with an error code of 442. As I am relatively new to Valhalla I don't know from which direction I should start debugging. I know about the major building blocks of Valhalla but how they interconnect and communicate with each other is still a mystery to me. So any guidance to debug it further would be really helpful. 

Best Regards

**Comment by nilsnolde** [comment]
Puuh, that's a VERY broad question. Let's take it over to discussions.

There's not much to recommend, other than setting up your dev environment properly (i.e. with breakpoint support) and read/understand the code between entrypoint (e.g. valhalla_service) and desired output. It's a lot to take in, but I can't see how we can support on such general stuff. It'll be easier once you have enough insight to ask detailed specific questions on concept/code API etc.

Or save yourself the trouble and ping us on enquiry@gis-ops.com, to have it done for you:)

**Reply by harvetech** [reply]
Thankyou @nilsnolde I know its a broad question but I had to start this discussion from somewhere. I think the problem is somewhere getting the right inputs for the algorithm e.g. I get this error messege. 

```
2023/06/06 15:25:58.159322 [INFO] algorithm::time_dependent_forward_a*
2023/06/06 15:25:58.170935 [ERROR] Route failed after iterations = 4
```

So I assume that the algo is not getting the correct input. I think the problem is related to **KTaxiAccess** which is the input to the TaxiCost class. I just copied that class and ofc changed this variable to the new one. But If I just take KTaxiAccess then it works. I think the problem is somewhere in a variable **KNewProfileAccess**.  On the other hand /locate does say the new profile has access and I think this variable only controls the access. 

I would be really interested in knowing how would you setup your dev environment ? so far I just change the files and build the docker and then see if it works or not. Am sure there is another way of testing it. Sorry for these basic questions.
